inline-caching/README.md

```diff
@@ -0,0 +1,247 @@
+You can either download postgres from https://www.postgresql.org/download/ and install it manually or if on a Mac, use brew install.
+
+```bash
+brew install postgresql
+brew services start postgresql
+```
+
+### Configure Postgres
+
+Now that postgres is running, we will want to create some tables and a user. One of the easier ways to do this
+is to use the test database that postgres provides for benchmarking via `pgbench`.
+
+```bash
+pgbench -i -s 50 postgres
+```
+
+This will create a few tables for us that we can manipulate. We will also need a user that our GemFire client can use to access the database
+
+```bash
+psql postgres
+create user myuser with encrypted password 'mypass';
+grant all privileges on database postgres to myuser;
+grant all privileges on all tables in schema public to myuser;
+```
+
+## Configure GemFire
+
+Once GemFire is installed, you need to configure GemFire to use Postgres as a data source. 
+This involves starting a GemFire cluster with all of the necessary class files as well as creating regions to store cached data and the data source to be used.
+
+This `cache-loader` project contains an implementation of `AsyncEventListener` and `CacheLoader` interfaces. 
+GemFire's AsyncEventListener and CacheLoader are useful features that can help improve the performance and scalability of your application by allowing you to asynchronously process write events and read data within your GemFire cluster.
+
+### AsyncEventListener
+The AsyncEventListener interface provides a way for your application to send write events to your database asynchronously in a batch operation, without blocking the sender. 
+This can help improve the throughput and performance of your application by allowing it to process events in parallel, while the sender continues to send more write events.
+This also helps keep the data in GemFire in sync with what is in the database.  
+
+In order to simplify the SQL necessary to write data to Postgres, the [JOOQ library](https://www.jooq.org/) is used. 
+
+```java
+int result = create.update(table)
+                    .set(filler, value)
+                    .where(tid.eq(itemId))
+                    .execute();
+```
+
+The table `pgbench_tellers` and column `filler` is something we get when postgres created our benchmark database. 
+It's just a place to store data for our example.
+
+We are getting our postgres credentials from System properties
+```java
+		String userName = System.getProperty("postgres.username");
+		String password = System.getProperty("postgres.password");
+```
+
+These properties will be passed in later when we create our GemFire region below.
+
+
+### CacheLoader
+GemFire's CacheLoader is used to fetch data from external systems and load it into the GemFire cache. 
+Our CacheLoader implementation also uses JOOQ for connecting to Postgres and Java system properties to retrieve Postgres' username and password.
+
+### AsyncEventListener
```

Comment:
We use the AsyncEventListener header twice in this section...are they same?  If not, how are they different? If they're different, the titles should indicate this somehow